option.go

```diff
@@ -240,6 +230,26 @@ func WithSendMaxBytes(max int) Option {
 	return &sendMaxBytesOption{Max: max}
 }
 
+// WithIdempotency declares the idempotency of the procedure. This can determine
+// whether a procedure call can safely be retried, and may affect which request
+// modalities are allowed for a given procedure call.
+//
+// In most cases, you should not need to manually set this. It is normally set
+// by the code generator for your schema.
```

Comment:
I'd include a code sample showing how to set idempotency level for a protobuf RPC. Connect's technically encoding-agnostic, but virtually everyone uses protobuf.

Reply:
Done. c98f933e4190e34e8ace8fa9955ee6c9728b0fc0